### PR TITLE
Add allow warnings flag to publish_private_pod

### PR DESCRIPTION
**Pull request by crazytonyli**
Similar to https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/82, this PR adds the `--allow-warnings` flag to another cocoapod command `publish_private_pod`, which we may need to use in some libraries.

---

- [x] I have considered if this change warrants release notes and have added them to the appropriate section in the `CHANGELOG.md` if necessary.

### CODE REVIEW

File: bin/publish_private_pod [hunk 1]
```diff
@@ -1,10 +1,15 @@
 #!/bin/bash -eu
 
 PATCH_COCOAPODS="false"
+COCOAPODS_FLAGS=(--verbose)
 
 while [[ "$#" -gt 0 ]]; do
 	case $1 in
-		--patch-cocoapods) PATCH_COCOAPODS="true" ;;
+		--patch-cocoapods)
+			PATCH_COCOAPODS="true" ;;
```
Comment (AliSoftware) [inline bin/publish_private_pod]: For stylistic consistency with rest of the code:
```suggestion
			PATCH_COCOAPODS="true"
			;;
```

**Review by AliSoftware — APPROVED** [review]

File: bin/publish_private_pod (hunk 1 above)
Comment (crazytonyli) [reply bin/publish_private_pod]: Good catch! Updated in https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/86/commits/a8e3224672945c82b326033156c577077ba5816d.

**Comment by AliSoftware** [comment]
Btw when I added similar changes to `publish_pod` I also added [some doc comments at the top of the scripts](https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/blob/trunk/bin/publish_pod#L3-L14)… maybe worth doing the same here?

File: bin/publish_private_pod (hunk 1 above)
Comment (AliSoftware) [reply bin/publish_private_pod]: Mmh seems that commit mixed tabs on line 9 vs spaces in line 10, which made the statements indentation misaligned (depending on your tab width settings)

**Comment by crazytonyli** [comment]
@AliSoftware Sure thing. Here is the follow up PR: https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/87

**Pull request merged**